### PR TITLE
node16 test fallbacks for old TS versions, fix dependency/devDependency import

### LINKED ISSUE #772: Allow imports of deps listed both in devDependencies and in dependencies
I was trying to lock down the version of `marked` for testing `marked-terminal` by specifying a specific version in `devDependencies` while `dependencies` had a large range:

```
    "dependencies": {
        "@types/cardinal": "^2.1",
        "@types/node": "*",
        "chalk": "^5.3.0",
        "marked": ">=6.0.0 <10"
    },
    "devDependencies": {
        "@types/marked-terminal": "workspace:.",
        "marked": "6.0.0"
    },
```

but @definitelytyped/no-import-of-dev-dependencies complains about imports of `marked` after this.

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #772 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67090#discussion_r1367525696

### CODE REVIEW

**Comment by andrewbranch** [comment]
I looked at the tests for the eslint rule and had no idea how to add to it because I didn’t see any package.json content.

File: packages/eslint-plugin/src/rules/no-import-of-dev-dependencies.ts [hunk 1]
```diff
@@ -26,16 +26,19 @@ const rule = createRule({
       const devdeps = packageJson
         ? Object.keys(packageJson.devDependencies).map((dep) => dep.replace(/@types\//, ""))
         : [];
+      const deps = packageJson
+        ? Object.keys(packageJson.dependencies ?? {}).map((dep) => dep.replace(/@types\//, ""))
+        : [];
```
Comment (jakebailey) [inline packages/eslint-plugin/src/rules/no-import-of-dev-dependencies.ts:31]: Preexisting but isn't this wrong when the package is scoped?

Comment (andrewbranch) [reply packages/eslint-plugin/src/rules/no-import-of-dev-dependencies.ts:31]: Yep

Comment (jakebailey) [reply packages/eslint-plugin/src/rules/no-import-of-dev-dependencies.ts:31]: 😬

**Comment by sandersn** [comment]
> I looked at the tests for the eslint rule and had no idea how to add to it because I didn’t see any package.json content. Tested both of these by linking locally. Feel free to push tests while I’m gone if you want.

My memory is that I couldn't figure out how to test these either; another reason to figure out how to switch to an existing rule.

**Review by jakebailey — APPROVED** [review]
LGTM but needs changesets for the two changed packages.

**Comment by jakebailey** [comment]
> I looked at the tests for the eslint rule and had no idea how to add to it because I didn’t see any package.json content.

Oh, and on this is because we misuse the package.json for the plugin itself as a source of tests... Very silly, and my other PR also helps that too (if we port every test to a fixture).

**Review by jakebailey — APPROVED** [review]

**Pull request merged**